Fixed an issue with states coming from `model.createMachine` resulting in contexts of type `any` after type refinements (#2436)
statelyai/xstate: .changeset/warm-spies-cry.md

---
'xstate': patch
---

> Fixed an issue with states coming from `model.createMachine` resulting in contexts of type `any` after type refinements such as here:

Author: I'm super tired already and I have a feeling that I've written some hard-to-understand gibberish here. If you have any ideas on how I could rephrase this I would appreciate the help.

Reviewer: Maybe something like?

Fixed issue where, when using `model.createMachine`, state context was incorrectly inferred as `any` after refinement with `.matches(..)`, for example:

Author: Thank you very much for taking a stab at rewording this! Would you like to prepare this as a PR suggestion that I could just commit with a button click? The change would also be contributed to you that way which is nice :)

Reviewer 2: ```suggestion
Fixed an issue where, when using `model.createMachine`, state context was incorrectly inferred as `any` after refinement with `.matches(...)`, such as in here:
```

Using @johtso's suggestion
